### PR TITLE
Code coverage of gsumm.c

### PR DESCRIPTION
**Pull request by MichaelChirico**
coverage now complete for this file

### CODE REVIEW

File: inst/tests/tests.Rraw [hunk 1]
```diff
@@ -8096,17 +8096,25 @@ DT = data.table(x=sample(5, 100, TRUE),
...
+test(1594.13, DT[, lapply(.SD, prod, na.rm=TRUE), by=x], DT[, lapply(.SD, base::prod, na.rm=TRUE), by=x])
+test(1594.14, DT[, lapply(.SD, prod, na.rm=TRUE), by=x, verbose=TRUE], output="GForce optimized j to.*gprod")
 
-test(1594.4, DT[, lapply(.SD, sd, na.rm=FALSE), by=x], DT[, lapply(.SD, stats::sd, na.rm=FALSE), by=x])
-test(1594.5, DT[, lapply(.SD, sd, na.rm=TRUE), by=x], DT[, lapply(.SD, stats::sd, na.rm=TRUE), by=x])
-test(1594.6, DT[, lapply(.SD, sd, na.rm=TRUE), by=x, verbose=TRUE], output="GForce optimized j to.*gsd")
-
-test(1594.7, DT[, lapply(.SD, prod, na.rm=FALSE), by=x], DT[, lapply(.SD, base::prod, na.rm=FALSE), by=x])
-test(1594.8, DT[, lapply(.SD, prod, na.rm=TRUE), by=x], DT[, lapply(.SD, base::prod, na.rm=TRUE), by=x])
-test(1594.9, DT[, lapply(.SD, prod, na.rm=TRUE), by=x, verbose=TRUE], output="GForce optimized j to.*gprod")
```
Comment (MichaelChirico) [inline inst/tests/tests.Rraw]: these tests were just renumbered. New tests are `05-08`

File: src/gsumm.c [hunk 2]
```diff
@@ -940,8 +940,6 @@ SEXP gmedian(SEXP x, SEXP narmArg) {
 
 SEXP glast(SEXP x) {
 
-  if (!isVectorAtomic(x)) error("GForce tail can only be applied to columns, not .SD or similar. To get tail of all items in a list such as .SD, either add the prefix utils::tail(.SD) or turn off GForce optimization using options(datatable.optimize=1).");
```
Comment (MichaelChirico) [inline src/gsumm.c]: There was a contradiction in both `glast` & `gfirst` between this error branch and the `VECSXP` branch below. I decided to try and remove this error & see what happens, and all the tests pass, so I opted to resolve the contradiction by eliminating this branch.

File: src/gsumm.c [hunk 3]
```diff
@@ -1204,7 +1200,7 @@ SEXP gnthvalue(SEXP x, SEXP valArg) {
 SEXP gvarsd1(SEXP x, SEXP narm, Rboolean isSD)
 {
   if (!isLogical(narm) || LENGTH(narm)!=1 || LOGICAL(narm)[0]==NA_LOGICAL) error("na.rm must be TRUE or FALSE");
-  if (!isVectorAtomic(x)) error("GForce var/sd can only be applied to columns, not .SD or similar. To find var/sd of all items in a list such as .SD, either add the prefix stats::var(.SD) (or stats::sd(.SD)) or turn off GForce optimization using options(datatable.optimize=1). More likely, you may be looking for 'DT[,lapply(.SD,var),by=,.SDcols=]'");
+  if (!isVectorAtomic(x)) error("GForce var/sd can only be applied to columns, not .SD or similar. For the full covariance matrix of all items in a list such as .SD, either add the prefix stats::var(.SD) (or stats::sd(.SD)) or turn off GForce optimization using options(datatable.optimize=1). Alternatively, if you only need the diagonal elements, 'DT[,lapply(.SD,var),by=,.SDcols=]' is the optimized way to do this.");
```
Comment (MichaelChirico) [inline src/gsumm.c]: I thought "more likely" sounded wrong here because getting the full covariance matrix is indeed a common use case

**Comment by mattdowle** [comment]
Great!

**Pull request merged**